lib/board.rb

```diff
@@ -0,0 +1,34 @@
+class Board
+  attr_reader :symb, :counter, :status, :b, :symbols
+
+  def initialize
+    @b = Array.new(3) { |i| Array.new(3) { |j| ((i * 3 + j + 1)).to_s } }
+    @counter = 9
+    @symb = ''
+    @symbols = %w[X O]
+  end
+
+  def turn(symb, loc)
+    if !loc.to_s.match?(/[1-9]/) || loc.to_s.length != 1 || @b[(loc - 1) / 3][(loc - 1) % 3].match?(/[XO]/)
+      @status = false
+    else
+      @status = true
+      @b[(loc - 1) / 3][(loc - 1) % 3] = symb
+      if winner(@b)
+        @counter = 0
+        @symb = symb
+      end
+      @counter -= 1
+    end
+  end
```

Comment:
That method do many things. Please, separate it
1- Check if the user input is valid
2- Update the Board
3- Checking the winner

Create at least those 3 methods into the `Board` class and update your while loop in `main.rb`.
This will make your code clearer and easily understandable.